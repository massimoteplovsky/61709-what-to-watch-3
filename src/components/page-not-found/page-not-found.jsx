import React from "react";
import Header from "../header/header.jsx";
import Footer from "../footer/footer.jsx";

const PageNotFound = () => {
  return (
    <div
      className="user-page"
    >
      <Header additionalClass="user-page__head"/>
      <h1
        className="title"
        style={{
          margin: `150px 0`,
          textAlign: `center`,
          fontSize: `60px`
        }}
      >Sorry, this page not found</h1>
      <Footer/>
    </div>
  );
};

export default PageNotFound;
