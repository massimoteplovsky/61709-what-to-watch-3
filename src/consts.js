export const DEFAULT_GENRE = `All genres`;
export const FILM_TO_SHOW = 8;
export const RELATED_FILM_COUNT = 4;
export const AUTH = `AUTH`;
export const NO_AUTH = `NO_AUTH`;
export const BASE_URL = `https://htmlacademy-react-2.appspot.com`;
export const MIN_COMMENT_LENGTH = 50;
export const MAX_COMMENT_LENGTH = 400;
export const NOT_FAVORITE_FILM = 0;
export const FAVORITE_FILM = 1;
export const MAX_RATING = 5;
