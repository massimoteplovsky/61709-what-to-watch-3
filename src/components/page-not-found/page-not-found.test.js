import React from "react";
import renderer from "react-test-renderer";
import {PageNotFound} from "./page-not-found.jsx";
import {Router} from "react-router-dom";
import {Provider} from "react-redux";
import history from "../../history.js";
import configureMockStore from "redux-mock-store";
import {AUTH} from "../../consts.js";

const mockStore = configureMockStore([]);
let store = mockStore({
  user: {
    authorizationStatus: AUTH
  }
});

it(`<PageNotFound /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PageNotFound/>
          </Router>
        </Provider>
    );

  expect(tree).toMatchSnapshot();
});
