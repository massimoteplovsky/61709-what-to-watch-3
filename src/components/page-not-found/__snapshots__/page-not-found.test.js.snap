// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> component renders correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt={null}
          height="63"
          onClick={[Function]}
          src={null}
          width="63"
        />
      </div>
    </div>
  </header>
  <h1
    className="title"
    style={
      Object {
        "fontSize": "60px",
        "margin": "150px 0",
        "textAlign": "center",
      }
    }
  >
    Sorry, this page not found
  </h1>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
